### DISCUSSION TITLE
Loading larger ISO files

### DISCUSSION THREAD

**Discussion by fre4ki** [ipxe/ipxe#476] [Q&A]
Hey guys,

is there a way for loading larger ISOs through http?

I have tried memdisk but it can't handle larger files (one with ~ 2.7 GB and one with ~ 12 GB). I had also no luck with SANBOOT and HTTP. Small Images like VMware ESXi works well.

Primary usage are servers with 64 GB+  RAM.

I'm using legacy BIOS.

Kind regards

**Comment by NiKiZe** [comment]
First of, what is on the ISO(s)?

And when it fails, what messages do you get? Is it maybe similar to "No space left on device (https://ipxe.org/34182000)" and if so, did you read that page and linked resources? (last section)

You are aware that most OSes on ISOs will not be bootable, as explained in the [iPXE forum FAQ](https://forum.ipxe.org/showthread.php?tid=5948)

> # I'm using memdisk or sanboot to boot an ISO and bootup fails
> In pcbios mode there is INT 13 which is used for disk access, this works fine before any kernel has started.
As soon as the kernel starts up, the real-mode INT 13 interface provided by iPXE's "sanboot" or memdisk ceases to exist.

> So first part of boot when kernel and initrd is loaded into memory works fine since that is using above mentioned INT 13. But when the kernel has started there no longer is any way to find that data.
You need to find a different method to provide the system with that data, most Linux distros support some kind of kernel cmdline option to be able to fetch squashfs image over NFS, FTP or HTTP (some have buggy DNS resolution)

> When using memdisk there is also the option of searching for and creating a device for the data, but it requires modification of the ISO, and in that case I would recommend to modify it to allow the squashfs to be loaded as an extra initrd instead, See this patch for SystemRescueCd

If you really really want to boot ISOs, then you should probably consider iSCSI and sanboot, note that it has to be full iSCSI, and the booted operating system need to have iSCSI boot support as well.
But for most scenarios there is better ways.
For Linuxes extract kernel and initrd and do proper netbooting, for windows use wimboot, etc.

**Reply by fre4ki** [reply]
Thanks for your reply. The ISO files are "HPE Smart Start" and "Lenovo XClarity". Both seems to be live DVDs and update the firmware of all server components.

I get an message like "no space left".

I see the article but sanboot with http was not working for me.

Is it better to sanboot with nfs rather than with http?

I'm sorry for the questions but i'm new at pxe. :-)

**Reply by fre4ki** [reply]
For Lenovo i created a pxe-boot folder with the following files:

![image](https://user-images.githubusercontent.com/62788741/134418626-9c7f1df5-509a-4a2b-8aea-1ee13c06a7e3.png)

I have a PXE File with this content:

![image](https://user-images.githubusercontent.com/62788741/134419526-64575685-d594-47c0-97c4-4c1ebeeef823.png)

It fails here:

![image](https://user-images.githubusercontent.com/62788741/134419687-7d50e04f-ca0f-432d-8693-c49297774b94.png)

Maybe i have to insert the squashfs.img file somewhere?

**Comment by NiKiZe** [comment]
Post the exact full message if yo want help.

I wrote that you could try iSCSI, Why would you think that NFS would work?

Yes the squashfs is needed, but how is up to the initscripts, which in some cases needs patching to find or download the files.
Example on use on Gentoo livecd: https://github.com/NiKiZe/Gentoo-iPXE/blob/main/combined.ipxe

**Reply by fre4ki** [reply]
Here is the full output from the last screen:

![image](https://user-images.githubusercontent.com/62788741/134523399-e17e05c5-076f-4605-ae5e-2fd1a2d3d3af.png)

I wrote about NFS because your linked post handles about that (https://forum.ipxe.org/showthread.php?tid=11991).

It seems there is a CentOS Linux. I try your example with Gentoo.

**Reply by fre4ki** [reply]
It seems there missed a file (this is the rdsosreport.txt file):

![image](https://user-images.githubusercontent.com/62788741/134524447-6ac0c4e5-286b-47b2-882f-5ec6fdf15e02.png)

**Reply by antonym** [reply]
I've found more success just extracting and then leveraging the live boot scripts or dracut to further mount the rootfs.  You can also leverage initrd to download the ISO mounted into the initrd and then leverage the boot scripts to load it.  It really just depends on the implementation of the ISO.

For CentOS based distros with dracut, you might use iso-scan:
```
kernel http://url.com/vmlinuz iso-scan/filename=/target.iso <other kernel options>
initrd http://url.com/initrd.img
initrd http://url.com/target.iso /target.iso
```

For Debian live-boot based distro's you may use FINDISO= instead.  

If the specific links are public to the ISOs you are trying, toss them here and I can take a look when I have a sec.

**Reply by fre4ki** [reply]
> I've found more success just extracting and then leveraging the live boot scripts or dracut to further mount the rootfs. You can also leverage initrd to download the ISO mounted into the initrd and then leverage the boot scripts to load it. It really just depends on the implementation of the ISO.
> 
> For CentOS based distros with dracut, you might use iso-scan:
> 
> ```
> kernel http://url.com/vmlinuz iso-scan/filename=/target.iso <other kernel options>
> initrd http://url.com/initrd.img
> initrd http://url.com/target.iso /target.iso
> ```
> 
> For Debian live-boot based distro's you may use FINDISO= instead.
> 
> If the specific links are public to the ISOs you are trying, toss them here and I can take a look when I have a sec.

I tried it but now i have a Kernel panic - not syncing: VFS: Unable to mount root fs on unknown-block(0,0)
